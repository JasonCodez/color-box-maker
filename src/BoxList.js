import { useState } from "react";
import { v4 as uuid } from "uuid";
import NewBoxForm from "./NewBoxForm";
import Box from "./Box";

const BoxList = () => {
   
   const [boxes, setBoxes] = useState([]);
   const addItem = (backgroundColor, width, height) => {
      setBoxes(boxes => [...boxes, {id: uuid(), backgroundColor, width, height}]);
   }
   return (
      <div>
         <NewBoxForm addItem={addItem}/>
         <div>
            {boxes.map(({id, backgroundColor, width, height}) => <Box id={id} backgroundColor={backgroundColor} width={width} height={height} key={uuid()} />)}
         </div>
      </div>
   )
}

export default BoxList;