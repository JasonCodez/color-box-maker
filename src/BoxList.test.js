import {render, fireEvent} from "@testing-library/react" ;
import BoxList from "./BoxList";

it("renders without crashing", function () {
   render(<BoxList />);
});

it("matches snapshot", function () {
   const { asFragment } = render(<BoxList />);
   expect(asFragment()).toMatchSnapshot();
});

it("should add new box", function () {
   const {getByLabelText, queryByText} = render(<BoxList />);
   const input = getByLabelText("Background Color");
   const btn = queryByText("Add Box");
});