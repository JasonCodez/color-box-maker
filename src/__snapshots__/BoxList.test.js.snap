// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div>
      <h2>
        Add New Box
      </h2>
      <form>
        <label
          for="backgroundColor"
        >
          Background Color 
        </label>
        <input
          autocomplete="off"
          id="backgroundColor"
          name="backgroundColor"
          placeholder="background color"
          required=""
          type="text"
          value=""
        />
        <label
          for="width"
        >
          Width 
        </label>
        <input
          autocomplete="off"
          id="width"
          name="width"
          placeholder="width"
          required=""
          type="text"
          value=""
        />
        <label
          for="height"
        >
          Height 
        </label>
        <input
          autocomplete="off"
          id="height"
          name="height"
          placeholder="height"
          required=""
          type="text"
          value=""
        />
        <button>
          Add Box
        </button>
      </form>
    </div>
    <div />
  </div>
</DocumentFragment>
`;
